### PR TITLE
Allows cloning dependencies

### PR DESCRIPTION
**Pull request by jkotalik**
Follow up to #414. Now adds ability to have a URL to clone from. Also addresses feedback after PR.

### CODE REVIEW

**Comment by davidfowl** [comment]
Few questions:
- How does it fail if you don't have git installed?
- How does it fail if you use private repo that requires a password.
- If there's no tye.yaml in the root, does it fail after clone?

**Comment by jkotalik** [comment]
> How does it fail if you don't have git installed?

I'll add a check for that. It would fail calling ProcessUtils.RunAsync.

> How does it fail if you use private repo that requires a password.

Do you think capturing the output from git clone would be good enough here for this scenario? I feel like it would be hard to know exactly what

> If there's no tye.yaml in the root, does it fail after clone?

No, it follows the same conventions for slns, csprojs, etc if a tye.yaml isn't present in the root of the project.

**Comment by jkotalik** [comment]
I'm also thinking we may need to specify a path to the tye.yaml inside of the repository (ex if the tye.yaml isn't present in the root).

**Comment by davidfowl** [comment]
> I'll add a check for that. It would fail calling ProcessUtils.RunAsync.

We do this today for Docker, we check if it's installed and running. I'd suggest following the same pattern.

> Do you think capturing the output from git clone would be good enough here for this scenario? I feel like it would be hard to know exactly what

We should at least understand what it does. Maybe it'll prompt and because we aren't redirecting stdin it'll fail.

**Comment by jkotalik** [comment]
Errors are outputted correctly, example of repository not found.
```
Failed to clone repository https://github.com/jkotalik/TyeMultiRepoVotingz with exit code 128.
    Cloning into 'C:\Users\jukotali\AppData\Local\Temp\1nwk0keb.zrk\.tye\deps\vote'...
    remote: Repository not found.
    fatal: repository 'https://github.com/jkotalik/TyeMultiRepoVotingz/' not found
```

File: src/Microsoft.Tye.Core/DirectoryExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+
+namespace Microsoft.Tye
+{
+    public static class DirectoryExtensions
```
Comment (rynowak) [inline src/Microsoft.Tye.Core/DirectoryExtensions.cs:9]: :(

Comment (jkotalik) [reply src/Microsoft.Tye.Core/DirectoryExtensions.cs:9]: What name/stuff would you prefer?

**Pull request merged**